### PR TITLE
gnupg 2.4.0

### PR DESCRIPTION
**Pull request by cho-m**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/gnupg.rb [hunk 1]
```diff
@@ -34,32 +34,21 @@ class Gnupg < Formula
 
   uses_from_macos "sqlite", since: :catalina
 
-  on_linux do
-    depends_on "libidn"
-  end
```
Comment (cho-m) [inline Formula/gnupg.rb:39]: No idea where this is from. Didn't see dependency in Arch or Debian so trying without it.

File: Formula/gnupg.rb [hunk 2]
```diff
@@ -31,35 +30,31 @@ class Gnupg < Formula
...
-
-  # Fixes a build failure without ldap.
-  # Committed upstream, will be in the next release.
-  # https://dev.gnupg.org/T6239
-  patch do
-    url "https://dev.gnupg.org/rG7011286ce6e1fb56c2989fdafbd11b931c489faa?diff=1"
-    sha256 "407011d4ae9799f50008b431df60cd5b781dca0f572e956fd46245aa209af7e8"
+  on_macos do
+    depends_on "gettext"
   end
```
Comment (cho-m) [inline Formula/gnupg.rb:41]: Need to try this out locally to make sure there is no build dependency on `gettext`. CI won't catch it as `libidn2` still has unnecessary dependency (#118912). Usually, `gettext` is only needed when needing to (re)generate configure as it is used during autogen/bootstrap scripts. Arch has no dependency which is a good sign as they build from source tarball unlike Debian which uses git repo.

**Review by p-linnane — APPROVED** [review]

File: Formula/gnupg.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/gnupg.rb:39]: looks like the git blame does not work either 😆 
image

File: Formula/gnupg.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/gnupg.rb:41]: yeah, usually it should be build dep, but i did see gettext got used for runtime as wlel.

**Review by chenrui333 — APPROVED** [review]

File: Formula/gnupg.rb (hunk 2 above)
Comment (cho-m) [reply Formula/gnupg.rb:41]: On Linux, runtime use is rare as `glibc` provides libintl.

The ones that need a runtime dependency often shell out to the binaries. I recall `byobu` executing gettext command at runtime. Formulae that use `Locale::gettext` Perl package are another example.

Comment (cho-m) [reply Formula/gnupg.rb:41]: No issues building without `gettext` installed. Configure does check and finds everything in glibc
```
configure: checking for gettext
checking whether NLS is requested... yes
checking for msgfmt... no
checking for gmsgfmt... :
checking for xgettext... no
checking for msgmerge... no
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyCurrent... no
checking for GNU gettext in libc... yes
checking whether to use NLS... yes
checking where the gettext function comes from... libc
```

**Pull request closed without merging**